thingsboard_gateway/gateway/tb_gateway_service.py

```diff
@@ -984,8 +985,14 @@ def __convert_telemetry_to_ts(data):
                 if isinstance(item['ts'], int):
                     telemetry_with_ts.append({"ts": item["ts"], "values": {**item["values"]}})
                 else:
-                    log.warning('Data has invalid TS (timestamp) format! Using generated TS instead.')
-                    telemetry_with_ts.append({"ts": int(time() * 1000), "values": {**item["values"]}})
+                    try:
```

Comment:
Probably it will be better to add a parameter for dateFormat to converter and resolve this point in converters, because it may be configurable by user. Also, this module is responsible about final data formatting and sending it to TB, it is not designed to use configuration for data formatting or anything else.